src/Swarm/Game/Step.hs

```diff
@@ -1845,6 +1867,18 @@ execConst c vs s k = do
       , prettyText (Out (VCApp c (reverse vs)) s k)
       ]
 
+  findNearest name = do
```

Comment:
Some type signatures on `findNearest` and possibly some of its local variables (`sortedLocs`, `genDiamondSides`, etc) might be really helpful.  I spent a while trying to understand it which was made more difficult by having to do type inference in my head.